### PR TITLE
Create a sandbox for end-to-end linalg on tensors

### PR DESCRIPTION
**Pull request by nicolasvasilache**
experimental/runners is a sandbox for abstractions and transformations
aiming at making linalg on tensors executable end-to-end with batteries
included.

Such abstractions are staged here and aim at graduating to core once they
have matured.

Additional support should strive to be the absolute minimum set of abstractions
required to execute. For larger scale usage, proper integration into IREE
should be considered first.

Depends on LLVM @0ee4bf151c0985027e82ec0036a655d68b4d6c37.

### CODE REVIEW

**Review by stellaraccident — APPROVED** [review]
I'm not going to spend a bunch of time reviewing experimental stuff like this which you are using as a staging area (and am supportive of you doing so) -- if you'd like a more detailed presubmit review of the code itself, feel free to ask (me or someone else).

The one change you need to make: Your BUILD files reference internal repository paths and are going to cause trouble. For an initial submission, do one of:

* Delete them entirely and submit in a separate patch (this is the least likely to cause friction for people doing the branch merges, because BUILD files are special).
* Rename them to BUILD.bazel so that they only apply on the OSS side. This should not create any overhead for the branch merges while you work it out. Even with this, these are not functional as-is as they reference non-existant paths.
* Submit the BUILD files internally once this is merged and see if the automation handles them for you :)

File: experimental/runners/BUILD [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    ],
+)
+
+package_group(
+    name = "friends",
+    includes = [
+        ":internal",
+    ],
+    packages = [
+        "//babelfish/device/...",
```
[the remark this answers is not in the document]
Comment (nicolasvasilache) [reply experimental/runners/BUILD]: dropped the BUILD, will resubmit sanitized later.

**Comment by nicolasvasilache** [comment]
Thanks! I still need to tidy up a few things so that it is really usable but it is getting there.

**Comment by GMNGeoffrey** [comment]
So this includes file with the LLVM license. It also has messed up header include guards. Can we please have more discussion before adding things to `iree/experimental`? I know we've had these conversations before with ModelBuilder and just kind of rolled with it, but each time we do something like this it adds additional burden to the build cops. Does this really need to be in IREE? What are the implications of having the code in here not under `third_party/` with the LLVM license? I understand (extremely well) how much of a PITA trying to integrate across google3 and GitHub is, but I'm not convinced continuing to stick stuff in the IREE repository is the way forward. Could this be open source only? Could we create a separate `iree-team-messes-around-with-stuff` repository?

**Pull request merged**